### ISSUE TITLE
Enum.chunk_every doesn't chunk correctly

### ISSUE THREAD

**Issue by ftes** [elixir-lang/elixir#12416]
### Elixir and Erlang/OTP versions

Erlang/OTP 25 [erts-13.1.2] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit:ns]

Elixir 1.14.3 (compiled with Erlang/OTP 23)

### Operating system

macOS Ventura 13.0.1

### Current behavior

```elixir
# Expected:
Enum.chunk_every([1, 2, 3], 2, 1, [4])
# [[1, 2], [2, 3], [3, 4]]

# Unexpected:
Enum.chunk_every([1, 2, 3], 3, 1, 4..5)
# Actual:   [[1, 2, 3], [2, 3, 4]]
# Expected: [[1, 2, 3], [2, 3, 4], [3, 4, 5]]
```

The docs for `Enum.chunk_every/4` state:
> Returns list of lists containing count elements each, where each new chunk starts step elements into the enumerable.

### Expected behavior

When using `step = 1`, I expect a chunk to be emitted for every element of the enumerable.

```elixir
Enum.chunk_every([1, 2, 3], 3, 1, 4..5)
# [[1, 2, 3], [2, 3, 4], [3, 4, 5]]
```

**Comment by ftes** [comment]
See #12417

**Comment by josevalim** [comment]
This is a documentation issue. You get steps _while_ you get full lists. I will improve the docs.

**Comment by josevalim** [comment]
Closed in 3913506bc5aae86044252b3674f5578b83f93649.

**Comment by ftes** [comment]
Thank you!
